crates/ide_assists/src/handlers/extract_function.rs

```diff
@@ -994,6 +994,9 @@ fn format_replacement(ctx: &AssistContext, fun: &Function, indent: IndentLevel)
         }
     }
     format_to!(buf, "{}", expr);
+    if body_contains_await(&fun.body) {
```

Comment:
It doesn't matter too much, but we could pull this check up to avoid doing it twice.